clang/lib/AST/ItaniumMangle.cpp

```diff
@@ -4492,6 +4492,31 @@ void CXXNameMangler::mangleType(const ArrayParameterType *T) {
   mangleType(cast<ConstantArrayType>(T));
 }
 
+void CXXNameMangler::mangleType(const HLSLAttributedResourceType *T) {
```

Comment:
I assume the intent is that we're mangling this as a vendor-specific qualifier? If so, we should use `mangleVendorQualifier` to ensure that it is properly prefixed, and we should probably use more than one letter as the qualifier name.

Reply:
Updated to use `mangleVendorQualifier`.